plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlCastPushdown.java

```diff
@@ -0,0 +1,483 @@
+
+                        // No pushdown for real datatype to integral types
+                        new InvalidCastTestCase("c_infinity_real_1", "tinyint", "Out of range for tinyint: Infinity"),
+                        new InvalidCastTestCase("c_infinity_real_1", "smallint", "Out of range for smallint: Infinity"),
+                        new InvalidCastTestCase("c_infinity_real_1", "integer", "Out of range for integer: Infinity"),
+                        new InvalidCastTestCase("c_infinity_real_2", "tinyint", "Out of range for tinyint: -Infinity"),
+                        new InvalidCastTestCase("c_infinity_real_2", "smallint", "Out of range for smallint: -Infinity"),
+                        new InvalidCastTestCase("c_infinity_real_2", "integer", "Out of range for integer: -Infinity"),
+
+                        // No pushdown for double precision datatype to integral types
+                        new InvalidCastTestCase("c_infinity_double_1", "tinyint", "Out of range for tinyint: Infinity"),
+                        new InvalidCastTestCase("c_infinity_double_1", "smallint", "Out of range for smallint: Infinity"),
+                        new InvalidCastTestCase("c_infinity_double_1", "integer", "Out of range for integer: Infinity"),
+                        new InvalidCastTestCase("c_infinity_double_1", "bigint", "Unable to cast Infinity to bigint"),
+                        new InvalidCastTestCase("c_infinity_double_2", "tinyint", "Out of range for tinyint: -Infinity"),
+                        new InvalidCastTestCase("c_infinity_double_2", "smallint", "Out of range for smallint: -Infinity"),
+                        new InvalidCastTestCase("c_infinity_double_2", "integer", "Out of range for integer: -Infinity"),
+                        new InvalidCastTestCase("c_infinity_double_2", "bigint", "Unable to cast -Infinity to bigint")));
+    }
+
+    @Test
+    public void testCastRealInfinityValueToBigint()
+    {
+        assertThat(query("SELECT CAST(c_Infinity_real AS BIGINT) FROM %s".formatted(leftTable())))
+                .matches("VALUES (BIGINT '9223372036854775807'), (BIGINT '-9223372036854775808'), (null)")
+                .isNotFullyPushedDown(ProjectNode.class);
+    }
+
+    @Test
+    public void testCastPushdownWithForcedTypedToInteger()
+    {
+        // These column types are not supported by default by trino. These types are forced mapped to varchar.
+        assertThat(query("SELECT CAST(c_xml AS VARCHAR(20)) FROM %s".formatted(leftTable())))
+                .isNotFullyPushedDown(ProjectNode.class);
+        assertThat(query("SELECT CAST(c_xml AS VARCHAR(20)) FROM %s".formatted(leftTable())))
+                .isNotFullyPushedDown(ProjectNode.class);
+    }
+
+    @Override
+    protected List<CastTestCase> supportedCastTypePushdown()
+    {
+        return ImmutableList.of(
```

Comment:
We prefer ImmutableList builder since it can reduce diff when we add a new entry in the future.